### DISCUSSION TITLE
How to actually skip directories in golangci-lint?

### DISCUSSION THREAD

**Discussion by roshan-rubrik** [golangci/golangci-lint#2903] [Q&A]
Golangci-lint doesn't give option to actually skip directories. The **skip-dirs** option is not skipping the directory, It just skips the reporting issues from that directory. Is there way to actually skip directories?
I am aware of this [stackoverflow thread](https://stackoverflow.com/questions/59857704/golangci-lint-want-to-really-ignore-a-go-file-not-simply-analyse-it-and-supp). This provides an option to actually skip files. I want to skip directories. Adding  codeanalysis tag in each file is not a option for us. 

In gometalinter, we had option to actually exclude directories. Why did similar option is not given in the golangci-lint?

**Comment by maxiride** [comment]
Stumbling here after few years, I want to exclude code generated using go generate or other tools (sqlc, openapicodegen, etc). Should be a common scenario for many.

Is the Stack Overflow answer still relevant today?

**Reply by ldez** [reply]
First, read this:
- https://golangci-lint.run/docs/linters/false-positives/#exclude
- https://golangci-lint.run/docs/welcome/faq/#why-is-it-not-possible-to-skipignore-typecheck-errors

The reports from generated code are skipped if the files have a comment that follows the convention `^// Code generated .* DO NOT EDIT\.$`
https://pkg.go.dev/cmd/go#hdr-Generate_Go_files_by_processing_source

If you want to exclude non-Go code, use the `ignore` directive inside your `go.mod` (require min go.25)
https://go.dev/ref/mod#go-mod-file-ignore

**Reply by bombsimon** [reply]
Yes, build tags from the SO thread is valid.

For others ending up here, see #4360 and #5438 for some newer context and discussions.
